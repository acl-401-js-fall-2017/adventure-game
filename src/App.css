.App {
  text-align: center;
  height: 40em;
  width: 40em;
}


