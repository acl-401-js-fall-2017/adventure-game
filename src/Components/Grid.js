import React, { Component } from 'react';
import Terrain from './Terrain';
import Gnome from './Gnome';
import './styles/Grid.css';

class Grid extends Component {

  render() {
    const { 
      gridArray, 
    } = this.props;

    const flattenedGrid = gridArray
      .reduce((outputArr, row) => outputArr.concat(row))
      .map((gridElem, i) => {
        const land = (
          <Terrain
            key={i}
            i={i}
            terrain={gridElem.terrain}
            hasKey={gridElem.hasKey}
            isLocked={gridElem.isLocked}
          ></Terrain>
        );
        const gnome = (
          <Gnome
            key={i}
            terrainColor={gridElem.terrain.color}
          ></Gnome>
        );
        return gridElem.hasGnome ? gnome : land;
      });

    return (
      <div className="Grid">
        {flattenedGrid}
      </div>
    );
  }
}

export default Grid;