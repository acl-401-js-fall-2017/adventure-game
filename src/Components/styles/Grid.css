.Grid {
    height: 40em;
    width: 40em;
    margin: auto;
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
    grid-template-rows: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
}
