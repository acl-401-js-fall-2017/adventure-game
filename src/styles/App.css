.App {
  text-align: center; }

.App-logo {
  height: 80px;
  width: auto;
  border: 2px solid black; }

.App-header {
  background-color: #d2691e;
  height: 140px;
  padding: 5px 35px 40px 35px;
  color: black; }
  .App-header > h2 {
    font-style: italic; }

.App-title {
  font-family: New Century Schoolbook, serif;
  font-size: 1.5em; }

.App-intro {
  font-size: large; }

.challenge .mission {
  font-family: serif;
  font-style: italic;
  font-size: 2em; }

.challenge .message {
  font-family: sans-serif;
  font-weight: bold; }

#nextButton {
  font-weight: bolder;
  background-color: green;
  border-radius: 15px; }
  #nextButton:hover {
    background-color: greenyellow; }

#prevButton {
  font-weight: bolder;
  background-color: #dd9c0f;
  border-radius: 15px; }
  #prevButton:hover {
    background-color: #e7030f; }
