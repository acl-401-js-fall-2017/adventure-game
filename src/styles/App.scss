.App {
  text-align: center;
}

.App-logo {
  height: 80px;
  width: auto;
  border: 2px solid black;
}

.App-header {
  background-color: #d2691e;
  height: 140px;
  padding: 5px 35px 40px 35px;
  color: black;
  > h2 {
    font-style: italic;
  }
}


.App-title {
  font-family: New Century Schoolbook,
  serif;
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

.challenge{
  .mission {
    font-family: serif;
    font-style: italic;
    font-size: 2em;
  }
  .message {
    font-family: sans-serif;
    font-weight: bold;
  }
}

#nextButton {
  font-weight: bolder;
  background-color: green;
  border-radius: 15px;
  &:hover {
    background-color: greenyellow;
  }
}

#prevButton {
  font-weight: bolder;
  background-color: rgb(221, 156, 15);
  border-radius: 15px;
  &:hover {
    background-color: rgb(231, 3, 15);
  }
}

