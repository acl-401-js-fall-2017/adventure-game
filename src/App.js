import React, { Component } from 'react';
import Stage from './Stage';
import Player from './Player';
import stages from './stages';
import './App.css';

class App extends Component {
  constructor () {
    super();
    this.state = {
      stages,
      stage: stages[0],
      player: {
        name: 'Big Shaq',
        pockets: [],
        health: 25,
        dialog: [
          'Skrrat, skidi-kat-kat. The name’s Big Shaq and man’s not hot.',
          'Hold up let me light this tree right quick. puff puff puff. How do I get out of here anyway?',
          'Whatever donut, your nose  long like a garden hose anyway.'
        ]
      },

    };
  }
  render() {
    return (
      <div className="App">
        
      </div>
    );
  }
}

export default App;
