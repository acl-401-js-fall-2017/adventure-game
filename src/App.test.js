import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import Grid from './Components/Grid';

import cartes from './utils/maps';
const carte = cartes.firstQuest;

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('renders a snapshot', () => {
  let gridSetUp = [];
  gridSetUp = carte.map(row => {
    return row.map(elem => {
      return {
        terrain: elem, 
        hasGnome: false
      };
    });
  });
  
  let gridArray = gridSetUp;
  
  let gnomeStats = {
    health: 100,
    strength: 1,
    speed: 10,
    pos: {
      X: 9,
      Y: 9
    }
  };

  const component = renderer.create(
    <Grid gridArray={gridArray} gnomeStats={gnomeStats}>GRID</Grid>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});