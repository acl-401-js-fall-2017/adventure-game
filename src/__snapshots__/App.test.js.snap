// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="Grid"
>
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  >
    <img
      alt=""
      src="chalis.svg"
    />
  </div>
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "gray",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "gray",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "gray",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "gray",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "gray",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "brown",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "gray",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "blue",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "gray",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "gray",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "gray",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
  <div
    className="Terrain"
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  />
</div>
`;
